File: lab1/lab1_Java/src/Task1.java
import java.util.Scanner;

public class Task1 {
    public static void calculatePotentialEnergy() {
        Scanner scanner = new Scanner(System.in);

        float m = 0;
        boolean validInputM = false;
        while (!validInputM) {
            System.out.print("Введіть масу тіла (у кілограмах): ");
            if (scanner.hasNextFloat()) {
                m = scanner.nextFloat();
                validInputM = true;
            } else {
                System.out.println("Неправильний формат! Будь ласка, введіть число.");
                scanner.next(); // Очистка буфера вводу
            }
        }

        float h = 0;
        boolean validInputH = false;
        while (!validInputH) {
            System.out.print("Введіть висоту підняття (у метрах): ");
            if (scanner.hasNextFloat()) {
                h = scanner.nextFloat();
                validInputH = true;
            } else {
                System.out.println("Неправильний формат! Будь ласка, введіть число.");
                scanner.next(); // Очистка буфера вводу
            }
        }

        // Гравітаційне прискорення (приблизно 9.8 м/с^2)
        float g = 9.8f;
        float E = m * g * h;
        System.out.println("Потенціальна енергія тіла: " + E + " джоулів");
    }
}

File: lab1/lab1_Kotlin/src/task1.kt
import java.util.Scanner

fun calculatePotentialEnergy() {
    val scanner = Scanner(System.`in`)

    var m = 0.0f
    var validInputM = false
    while (!validInputM) {
        print("Введіть масу тіла (у кілограмах): ")
        if (scanner.hasNextFloat()) {
            m = scanner.nextFloat()
            validInputM = true
        } else {
            println("Неправильний формат! Будь ласка, введіть число.")
            scanner.next() // Очистка буфера вводу
        }
    }

    var h = 0.0f
    var validInputH = false
    while (!validInputH) {
        print("Введіть висоту підняття (у метрах): ")
        if (scanner.hasNextFloat()) {
            h = scanner.nextFloat()
            validInputH = true
        } else {
            println("Неправильний формат! Будь ласка, введіть число.")
            scanner.next() // Очистка буфера вводу
        }
    }

    // Гравітаційне прискорення (приблизно 9.8 м/с^2)
    val g = 9.8f
    val E = m * g * h
    println("Потенціальна енергія тіла: $E джоулів")
}

